components/printer/loading.go

```diff
@@ -0,0 +1,27 @@
+package printer
+
+import (
+	"time"
+
+	"github.com/briandowns/spinner"
+)
+
+type Func[T any] func() (T, error)
+
+var loadingSpinner *spinner.Spinner
+
+func init() {
```

Comment:
Who calls this `init` function?

Reply:
By go itself: https://stackoverflow.com/questions/24790175/when-is-the-init-function-run